java/test/jmri/jmrit/display/layoutEditor/LayoutEditorChecksTest.java

```diff
@@ -1,40 +1,349 @@
 package jmri.jmrit.display.layoutEditor;
 
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.Before;
-import org.junit.Test;
-
-import jmri.util.JUnitUtil;
-
 import java.awt.GraphicsEnvironment;
+import java.awt.geom.*;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.swing.*;
+import jmri.InstanceManager;
+import jmri.jmrit.display.EditorFrameOperator;
+import jmri.util.*;
+import jmri.util.junit.rules.RetryRule;
+import org.junit.*;
+import org.junit.rules.Timeout;
+import org.netbeans.jemmy.QueueTool;
+import org.netbeans.jemmy.operators.*;
 
 /**
- *
  * @author Paul Bender Copyright (C) 2017
+ * @author George Warner Copyright: (C) 2019
  */
 public class LayoutEditorChecksTest {
 
+    @Rule   // 10 second timeout for methods in this test class.
+    public Timeout globalTimeout = Timeout.seconds(10);
+
+    @Rule   // allow 5 retries
+    public RetryRule retryRule = new RetryRule(5);
+
+    //LayoutEditorChecks Bundle Strings
+    private String toolsMenuTitle = Bundle.getMessage("MenuTools");
+    private String checkMenuTitle = Bundle.getMessage("CheckMenuTitle");
+    private String checkUnConnectedTracksMenuTitle = Bundle.getMessage("CheckUnConnectedTracksMenuTitle");
+    private String checkUnBlockedTracksMenuTitle = Bundle.getMessage("CheckUnBlockedTracksMenuTitle");
+    private String checkNonContiguousBlocksMenuTitle = Bundle.getMessage("CheckNonContiguousBlocksMenuTitle");
+    private String checkUnnecessaryAnchorsMenuTitle = Bundle.getMessage("CheckUnnecessaryAnchorsMenuTitle");
+
+    //name strings
+    private static String myBlockName = "My Block";
+    private static String rightHandName = "Right Hand";
+    private static String leftHandName = "Left Hand";
+
+    //jemmy operators
+    private static EditorFrameOperator layoutEditorEFO = null;
+
+    private static LayoutEditor layoutEditor = null;
+    private static LayoutEditorChecks layoutEditorChecks = null;
+    private static LayoutBlock layoutBlock = null;
+    private static LayoutTurnout ltRH = null, ltLH = null;
+    private static PositionablePoint a1 = null;
+    private static PositionablePoint a2 = null;
+    private static TrackSegment ts1 = null;
+    private static TrackSegment ts2 = null;
+    private static TrackSegment ts3 = null;
+
     @Test
     public void testCTor() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        LayoutEditor le = new LayoutEditor("Layout Editor Checks Test Layout");
-        LayoutEditorChecks t = new LayoutEditorChecks(le);
-        Assert.assertNotNull("exists", t);
-        JUnitUtil.dispose(le);
+        Assert.assertNotNull("exists", layoutEditorChecks);
     }
 
-    // The minimal setup for log4J
-    @Before
-    public void setUp() {
+    @Test
+    public void testSetup() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
```

Comment:
I don’t think this test that the setup method completed belongs here. If the setup fails, then tests that use the configured components will fail.  That is the expected behavior.

Reply:
I can live with that… Fixed.